// @stjiris/jurisprudencia-document/dist/hash.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.calculateUUID = exports.calculateHASH = exports.calculateSHA1 = void 0;
const crypto_1 = require("crypto");
// Create a deterministic hash from stringifying an Record<string, string> object
function calculateSHA1(obj, keys) {
    let keysToUse = (keys === null || keys === void 0 ? void 0 : keys.slice(0)) || Object.keys(obj);
    keysToUse.sort();
    let str = JSON.stringify(obj, keysToUse);
    return (0, crypto_1.createHash)("sha1").update(str).digest("base64url");
}
exports.calculateSHA1 = calculateSHA1;
function calculateHASH(doc) {
    return {
        Original: calculateSHA1(doc.Original),
        Processo: calculateSHA1(doc["Número de Processo"] || ""),
        Sumário: calculateSHA1(doc.Sumário || ""),
        Texto: calculateSHA1(doc.Texto || "")
    };
}
exports.calculateHASH = calculateHASH;
function calculateUUID(hash) {
    return calculateSHA1(hash, ["Sumário", "Texto", "Processo"]);
}
exports.calculateUUID = calculateUUID;
