// @stjiris/jurisprudencia-document/dist/jurisprudencia-document.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isJurisprudenciaDocumentGenericKey = exports.isJurisprudenciaDocumentExactKey = exports.isJurisprudenciaDocumentTextKey = exports.isJurisprudenciaDocumentDateKey = exports.isJurisprudenciaDocumentObjectKey = exports.isJurisprudenciaDocumentHashKey = exports.isJurisprudenciaDocumentContentKey = exports.isJurisprudenciaDocumentStateKey = exports.isValidJurisprudenciaDocumentKey = exports.JurisprudenciaDocumentKeys = exports.JurisprudenciaDocumentGenericKeys = exports.JurisprudenciaDocumentExactKeys = exports.JurisprudenciaDocumentTextKeys = exports.JurisprudenciaDocumentDateKeys = exports.JurisprudenciaDocumentObjectKeys = exports.JurisprudenciaDocumentHashKeys = exports.JurisprudenciaDocumentContentKeys = exports.JurisprudenciaDocumentStateValues = exports.JurisprudenciaDocumentStateKeys = exports.JurisprudenciaDocumentProperties = exports.JurisprudenciaVersion = void 0;
exports.JurisprudenciaVersion = "jurisprudencia.12.0";
const ExactFieldMapping = {
    type: 'keyword'
};
const NormalizedFieldMapping = {
    type: 'keyword',
    normalizer: 'term_normalizer'
};
const DateFieldMapping = {
    type: 'date',
    format: 'dd/MM/yyyy'
};
const GenericFieldMapping = {
    properties: {
        Original: ExactFieldMapping,
        Show: ExactFieldMapping,
        Index: {
            type: "text",
            fielddata: true,
            fields: {
                raw: ExactFieldMapping,
                keyword: NormalizedFieldMapping
            }
        }
    }
};
const ObjectFieldMapping = {
    type: 'object',
    enabled: false
};
const TextFieldMapping = {
    type: 'text',
    term_vector: 'with_positions_offsets_payloads'
};
exports.JurisprudenciaDocumentProperties = {
    "Original": ObjectFieldMapping,
    "Número de Processo": NormalizedFieldMapping,
    "ECLI": ExactFieldMapping,
    "Data": DateFieldMapping,
    "Tipo": ExactFieldMapping,
    "Relator Nome Profissional": GenericFieldMapping,
    "Relator Nome Completo": GenericFieldMapping,
    "Descritores": GenericFieldMapping,
    "Meio Processual": GenericFieldMapping,
    "Votação": GenericFieldMapping,
    "Secção": GenericFieldMapping,
    "Área": GenericFieldMapping,
    "Decisão": GenericFieldMapping,
    "Tribunal de Recurso": GenericFieldMapping,
    "Tribunal de Recurso - Processo": GenericFieldMapping,
    "Área Temática": GenericFieldMapping,
    "Jurisprudência Estrangeira": GenericFieldMapping,
    "Jurisprudência Internacional": GenericFieldMapping,
    "Jurisprudência Nacional": GenericFieldMapping,
    "Doutrina": GenericFieldMapping,
    "Legislação Comunitária": GenericFieldMapping,
    "Legislação Estrangeira": GenericFieldMapping,
    "Legislação Nacional": GenericFieldMapping,
    "Referências Internacionais": GenericFieldMapping,
    "Indicações Eventuais": GenericFieldMapping,
    "Referência de publicação": GenericFieldMapping,
    "Jurisprudência": GenericFieldMapping,
    "Sumário": TextFieldMapping,
    "Texto": TextFieldMapping,
    "Fonte": ExactFieldMapping,
    "URL": ExactFieldMapping,
    "UUID": ExactFieldMapping,
    "HASH": {
        type: "object",
        properties: {
            "Original": { type: "keyword" },
            "Texto": { type: "keyword" },
            "Sumário": { type: "keyword" },
            "Processo": { type: "keyword" }
        }
    },
    "CONTENT": {
        type: 'text'
    },
    "STATE": ExactFieldMapping,
};
exports.JurisprudenciaDocumentStateKeys = ["STATE"];
exports.JurisprudenciaDocumentStateValues = ["importação", "preparação", "público", "privado", "eliminado"];
exports.JurisprudenciaDocumentContentKeys = ["CONTENT"];
exports.JurisprudenciaDocumentHashKeys = ["HASH"];
exports.JurisprudenciaDocumentObjectKeys = ["Original"];
exports.JurisprudenciaDocumentDateKeys = ["Data"];
exports.JurisprudenciaDocumentTextKeys = ["Sumário", "Texto"];
exports.JurisprudenciaDocumentExactKeys = ["Número de Processo", "ECLI", "Fonte", "URL", "UUID", "Tipo"];
exports.JurisprudenciaDocumentGenericKeys = ["Relator Nome Profissional", "Relator Nome Completo", "Descritores", "Meio Processual", "Votação", "Secção", "Área", "Decisão", "Tribunal de Recurso", "Tribunal de Recurso - Processo", "Área Temática", "Jurisprudência Estrangeira", "Jurisprudência Internacional", "Jurisprudência Nacional", "Doutrina", "Legislação Comunitária", "Legislação Estrangeira", "Legislação Nacional", "Referências Internacionais", "Indicações Eventuais", "Referência de publicação", "Jurisprudência"];
exports.JurisprudenciaDocumentKeys = Object.keys(exports.JurisprudenciaDocumentProperties);
function isValidJurisprudenciaDocumentKey(accessKey) {
    return accessKey in exports.JurisprudenciaDocumentProperties;
}
exports.isValidJurisprudenciaDocumentKey = isValidJurisprudenciaDocumentKey;
function isJurisprudenciaDocumentStateKey(accessKey) {
    return isValidJurisprudenciaDocumentKey(accessKey) && exports.JurisprudenciaDocumentStateKeys.includes(accessKey);
}
exports.isJurisprudenciaDocumentStateKey = isJurisprudenciaDocumentStateKey;
function isJurisprudenciaDocumentContentKey(accessKey) {
    return isValidJurisprudenciaDocumentKey(accessKey) && exports.JurisprudenciaDocumentContentKeys.includes(accessKey);
}
exports.isJurisprudenciaDocumentContentKey = isJurisprudenciaDocumentContentKey;
function isJurisprudenciaDocumentHashKey(accessKey) {
    return isValidJurisprudenciaDocumentKey(accessKey) && exports.JurisprudenciaDocumentHashKeys.includes(accessKey);
}
exports.isJurisprudenciaDocumentHashKey = isJurisprudenciaDocumentHashKey;
function isJurisprudenciaDocumentObjectKey(accessKey) {
    return isValidJurisprudenciaDocumentKey(accessKey) && exports.JurisprudenciaDocumentObjectKeys.includes(accessKey);
}
exports.isJurisprudenciaDocumentObjectKey = isJurisprudenciaDocumentObjectKey;
function isJurisprudenciaDocumentDateKey(accessKey) {
    return isValidJurisprudenciaDocumentKey(accessKey) && exports.JurisprudenciaDocumentDateKeys.includes(accessKey);
}
exports.isJurisprudenciaDocumentDateKey = isJurisprudenciaDocumentDateKey;
function isJurisprudenciaDocumentTextKey(accessKey) {
    return isValidJurisprudenciaDocumentKey(accessKey) && exports.JurisprudenciaDocumentTextKeys.includes(accessKey);
}
exports.isJurisprudenciaDocumentTextKey = isJurisprudenciaDocumentTextKey;
function isJurisprudenciaDocumentExactKey(accessKey) {
    return isValidJurisprudenciaDocumentKey(accessKey) && exports.JurisprudenciaDocumentExactKeys.includes(accessKey);
}
exports.isJurisprudenciaDocumentExactKey = isJurisprudenciaDocumentExactKey;
function isJurisprudenciaDocumentGenericKey(accessKey) {
    return isValidJurisprudenciaDocumentKey(accessKey) && exports.JurisprudenciaDocumentGenericKeys.includes(accessKey);
}
exports.isJurisprudenciaDocumentGenericKey = isJurisprudenciaDocumentGenericKey;
